bin/i18n/redact_restore_utils.rb

```diff
@@ -57,70 +68,54 @@ def self.restore_data(source_data, redacted_data, plugins = [], format = 'md')
     source_json.close
     redacted_json.close
 
-    return restored
+    return JSON.parse(restored)
   end
 
   def self.restore(source, redacted, dest, plugins = [], format = 'md')
     return unless File.exist?(source)
     return unless File.exist?(redacted)
-    is_json = File.extname(source) == '.json'
-    source_data =
-      if is_json
-        JSON.parse(File.read(source))
-      else
-        YAML.load_file(source)
-      end
-    redacted_data =
-      if is_json
-        JSON.parse(File.read(redacted))
-      else
-        YAML.load_file(redacted)
-      end
 
-    return unless source_data
-    return unless redacted_data
-    return unless source_data&.values&.first&.length
-    return unless redacted_data&.values&.first&.length
+    if I18nScriptUtils.json_file?(source) || I18nScriptUtils.yaml_file?(source)
+      source_data = I18nScriptUtils.parse_file(source)
+      redacted_data = I18nScriptUtils.parse_file(redacted)
 
-    restored =
-      if is_json
-        RedactRestoreUtils.restore_data(source_data, redacted_data, plugins, format)
-      else
-        RedactRestoreUtils.restore_data(source_data.values.first, redacted_data.values.first, plugins, format)
-      end
+      return unless source_data
+      return unless redacted_data
+      return unless source_data&.values&.first&.length
+      return unless redacted_data&.values&.first&.length
```

Comment:
I'm not sure `.length` is doing anything here. When would it be `false` when `source_data.values.first` exists?

Reply:
It was in the code before, so I did not want to mess with it in case it would cover an edge case.

Reply:
It was there before, so let's not mess with it! (but there is no 'falsey' length value, so I'm not sure what it does)

Reply:
My best guess is that if we have an object like this: `en => {}`, where we have the locale key, but no content inside (no translations).